source/pr_cmds.c

```diff
@@ -1487,330 +1487,286 @@ This is where the magic happens
 =================
 */
 
-int closedset[MAX_WAYPOINTS]; // The set of nodes already evaluated.
-int openset[MAX_WAYPOINTS];//Actual sorted open list
-int opensetRef[MAX_WAYPOINTS];//Reference values of open list
-int opensetLength;//equivalent of javaScript's array[].length;
+
 #define MaxZombies 16
 
+
+#define WAYPOINT_SET_NONE 	0
+#define WAYPOINT_SET_OPEN 	1
+#define WAYPOINT_SET_CLOSED	2
+
+
+char waypoint_set[MAX_WAYPOINTS]; // waypoint_set[i] contains the set identifier for the i-th waypoint
+unsigned short openset_waypoints[MAX_WAYPOINTS]; // List of waypoints currently in the open set sorted by heuristic cost (index 0 contains lowest cost waypoint)
+unsigned short openset_length; // Current length of the open set
 zombie_ai zombie_list[MaxZombies];
```

Comment:
We should use the same value returned by the PF_maxai builtin.

Reply:
Looks like `nzp_maxai` maps to `void PF_MaxZombies(void)` which has a hardcoded `12

I partly think we should rename things to be zombie-agnostic (for other AI types), but navmeshes will eventually replace it all.

I think for now the best move might be to update `PF_MaxZombies` to instead return this `MaxZombies` preprocessor define.

Reply:
Sure. To clarify, my concern was the disparity between these numbers. If we can change this macro to 12 and have this builtin return this macro as well, that is perfectly fine.